### PR TITLE
Restrict node versions

### LINKED ISSUE #287: Upgrade default Node version to current LTS (14.x), drop support for versions < 10.

### PR DESCRIPTION
**Pull request by davemcorwin**
Resolves #287 

@amirbey @apburnes Wanted to get y'alls thoughts on this. Sets the default Node version to 12 (seemed like its probably the one most folks are using) and restricts usage to only LTS major versions (10, 12, 14), but allows install and use of any version for those majors. I'm trying to strike a balance between enforcement and flexibility.

Tests are still a TODO

### CODE REVIEW

**Review by amirbey — APPROVED** [review]

**Comment by apburnes** [comment]
Looks good. Would it be make sense to start out with only throwing a warning and not failing the build so we can announce the deprecation of older node versions? Also, is there a way we could log which sites are running older versions so we can target outreach about updating the node versions and provide documentation?

**Comment by davemcorwin** [comment]
> Looks good. Would it be make sense to start out with only throwing a warning and not failing the build so we can announce the deprecation of older node versions? 

Yes, but most people wouldn't see it since folks only look at the logs when there is a problem. We've considered surfacing warnings as additional statuses on the branch though it's unclear how much more visibility that would provide.

>Also, is there a way we could log which sites are running older versions so we can target outreach about updating the node versions and provide documentation?

A little bit ago I investigated this for sites with custom domains and I'm trying to track it down. I think there were only a couple still on Node 8.

**Comment by apburnes** [comment]
Since warnings could probably be missed by our users, should we take this opportunity to post the deprecation notice to status page for node < v10?  If their builds failed, we can point to our status page notification.

**Comment by davemcorwin** [comment]
oh snap, that's a great idea!!!

File: Dockerfile [hunk 1]
```diff
@@ -20,14 +20,12 @@ ENV LC_ALL en_US.UTF-8
 
 RUN dpkg-reconfigure --frontend noninteractive locales
 
-# Install nvm and install versions 8 and 10
+# Default to Node 12 (erbium)
 ENV NVM_DIR /usr/local/nvm
-ENV NODE_DEFAULT_VERSION 10
-RUN curl https://raw.githubusercontent.com/creationix/nvm/v0.31.3/install.sh | bash \
+RUN mkdir $NVM_DIR \
+  && curl https://raw.githubusercontent.com/nvm-sh/nvm/v0.37.0/install.sh | bash \
```
Comment (davemcorwin) [inline Dockerfile:26]:
```suggestion
  && curl https://raw.githubusercontent.com/nvm-sh/nvm/v0.37.2/install.sh | bash \
```

File: Dockerfile-exp [hunk 2]
```diff
@@ -38,13 +38,12 @@ RUN \
   && rvm use --default $RUBY_VERSION \
   && echo 'rvm_silence_path_mismatch_check_flag=1' >> /etc/rvmrc
 
+# Default to Node 12 (erbium)
 ENV NVM_DIR /usr/local/nvm
-ENV NODE_DEFAULT_VERSION 10
-RUN \
-  curl https://raw.githubusercontent.com/creationix/nvm/v0.31.3/install.sh | bash \
+RUN mkdir $NVM_DIR \
+  && curl https://raw.githubusercontent.com/nvm-sh/nvm/v0.37.0/install.sh | bash \
```
Comment (davemcorwin) [inline Dockerfile-exp:44]:
```suggestion
  && curl https://raw.githubusercontent.com/nvm-sh/nvm/v0.37.2/install.sh | bash \
```

**Comment by davemcorwin** [comment]
From what I can tell, I don't think we have an production customers using < node 10 so I think this is ok to merge into staging. I will add a deprecation notice on statuspage just in case!

**Pull request merged**